contribute/CONTRIBUTING-TO-CI-AND-E2E.md

```diff
@@ -1,112 +1,99 @@
-
-### Raise or fix issues on the CI framework 
-
-As with other issues, please mention the bug/enhancements in detail with logs/references if possible.
-
-The jenkins configuration used to setup OpenEBS e2e can be found [here](https://github.com/openebs/openebs/blob/master/e2e/jenkins/README.md)
-
-### Raise or fix issues on the e2e, CI documentation
-
-Getting Documentation Right is Hard! Please raise a PR with you proposed changes to the READMEs, tutorials and deployment guides in openebs/e2e
-
-# HAPPY TESTING !!
-
-
-
-  
-
-
-
-
-
-
-
-
-
+## INTRODUCTION TO OPENEBS E2E
+
+OpenEBS e2e is focused on workload simulation and application deployment on OpenEBS storage, predominantly in a Kubernetes environment. In its current state, it includes application workflow tests, basic compliance and resiliency tests. The scope of the said tests is expected to evolve over time to include performance and security tests as well.
+ 
+E2e is mostly written in Ansible, that is, as Ansible playbooks with docker used for test images (these include standard application images from dockerhub as well as custom-built images stored in openebs/test-storage). 
+
+Since the tests are performed in a Kubernetes environment, the test images are typically deployed as "pods" and "jobs". Therefore they are accompanied by the respective YAML specification files.
+
+## BUILDING BLOCKS OF OPENEBS CI
+
+OpenEBS Continuous Integration (CI) has two distinct parts:
+
+- Travis is used to run unit tests, build code and push images to dockerhub (in case of repositories where the image is the build artifact).
+- Jenkins pulls these images, creates the Kubernetes-based test environment and executes e2e. It is also used to run tests on commits that change application specifications or OpenEBS Kubernetes deployment specifications.
+
+While you can run OpenEBS e2e as a standalone suite, the priority is to run them on commits done on the core OpenEBS Github repositories such as openebs/openebs, openebs/mayaserver, openebs/jiva. This part of the continuous integration is built around a dedicated Jenkins server. On detecting commits via a polling mechanism, Jenkins launches VMs using Vagrant and runs the Ansible playbooks to setup the environment and execute e2e tests. The openebs/e2e project focuses on this part of the CI.
+
+The following schema illustrates the workflow:-
+
+![OpenEBS CI Overview Diagram](https://Github.com/ksatchit/openebs/blob/master/documentation/source/_static/OpenEBS_CI_Workflow.png)
+
+## CONTRIBUTING TO E2E and OPENEBS CI 
+
+Contributions are welcome!!  Before starting, read the project documentation at [https://docs.openebs.io/](https://docs.openebs.io/) is highly recommended. Please raise a Github issue for any questions you may have on the project in general or e2e or CI in particular.
```

Comment:
read the project = reading the project

Reply:
There is already a verb before that, it would look very bad there.